docs/architecture/grpc-for-wcf-developers/application-performance-management.md

```diff
@@ -0,0 +1,166 @@
+---
+title: Application Performance Management - gRPC for WCF Developers
+description: Logging, metrics and tracing for ASP.NET Core gRPC applications.
+author: markrendle
+ms.date: 09/02/2019
+---
+
+# Application Performance Management
+
+In modern production environments like Kubernetes, it's very important to be able to monitor applications to ensure they're running optimally. Concerns like logging and metrics have never been more important. ASP.NET Core, including gRPC, has first-class support for producing and managing log messages and metrics data, as well as *tracing* data. This section will explore these areas in more details.
+
+## Logging vs Metrics
+
+Before looking at the implementation details, it's necessary to understand the difference between logging and metrics.
+
+Logging is concerned with text messages that record detailed information about things that have happened in the system. Log messages may include exception data like stack traces, or structured data that provides context about the message. Logging output is commonly written to a searchable text store.
+
+Metrics refers to numeric data that is designed to be aggregated and presented using charts and graphs in a dashboard that provides a view of the overall health and performance of an application. Metrics data can also be used to trigger automated alerts when a threshold is exceeded. The following list shows some examples of metrics data:
+
+- Time taken to process requests.
+- The number of requests per second being handled by an instance of a service.
+- The number of failed requests on an instance.
+
+## Logging in ASP.NET Core gRPC
```

Comment:
Should talk about logging in the client somewhere. It has a GrpcClientOptions.LoggerFactory option. Can be set manually with a LoggerFactory resolved from DI, or the HttpClientFactory integration will set the logger factory for you.